examples/nlp/abstractive_summarization_with_bart.py

```diff
@@ -0,0 +1,249 @@
+"""
+Title: Abstractive Text Summarization with BART
+Author: [Abheesht Sharma](https://github.com/abheesht17/)
+Date created: 2023/07/08
+Last modified: 2023/07/08
+Description: Use KerasNLP to fine-tune BART on the abstractive summarization task.
+Accelerator: GPU
+"""
+
+"""
+## Introduction
+
+In the era of information overload, it has become crucial to extract the crux
+of a long document or a conversation and express it in a few sentences. Owing
+to the fact that summarization has widespread applications in different domains,
+it has become a key, well-studied NLP task in recent years.
+
+[Bidirectional Autoregressive Transformer (BART)](https://arxiv.org/abs/1910.13461)
+is a Transformer-based encoder-decoder model, often used for
+sequence-to-sequence tasks like summarization and neural machine translation.
+BART is pre-trained in a self-supervised fashion on a large text corpus. During
+pre-training, the text is corrupted and BART is trained to reconstruct the
+original text (hence called a "denoising autoencoder"). Some pre-training tasks
+include token masking, token deletion, sentence permutation (shuffle sentences
+and train BART to fix the order), etc.
+
+In this example, we will demonstrate how to fine-tune BART on the abstractive
+summarization task (on conversations!) using KerasNLP, generate summaries using
+the fine-tuned model, and evaluate the summaries using ROUGE score.
+"""
+
+"""
+## Setup
+
+Before we start implementing the pipeline, let's install and import all the
+libraries we need. We'll be using the KerasNLP library. We will also need a
+couple of utility libraries.
+"""
+
+"""shell
+pip install rouge-score -q
+pip install git+https://github.com/abheesht17/keras-nlp.git@fix-bart-mp -q
+pip install py7zr -q
+pip install gdown -q
+"""
+
+import gdown
+import py7zr
+import time
+
+import keras_nlp
+import tensorflow as tf
+import tensorflow_datasets as tfds
+
+from tensorflow import keras
+
+"""
+Let's also define our hyperparameters.
+"""
+
+BATCH_SIZE = 16
+NUM_BATCHES = 500
+EPOCHS = 1  # Can be set to a higher value for better results
+MAX_ENCODER_SEQUENCE_LENGTH = 512
+MAX_DECODER_SEQUENCE_LENGTH = 128
+MAX_GENERATION_LENGTH = 40
+
+"""
+## Dataset
+
+Let's load the [SAMSum dataset](https://arxiv.org/abs/1911.12237). This dataset
+contains around 15,000 pairs of conversations/dialogues and summaries.
+"""
+
+# Download the dataset.
+gdown.download(
+    "https://drive.google.com/uc?id=1u4L2tgfbd5RLYPws47LTE9GeIJ6bZA9g",
```

Comment:
where is this drive link from? can we not use keras.utils.get_file?

Reply:
Followed the manual download instructions here: https://www.tensorflow.org/datasets/catalog/samsum. The provided link intermittently fails with `wget` (and `keras.utils.get_file`, by extension). I uploaded it to my GDrive. `keras.utils.get_file` might not work on a GDrive link (wget wasn't working), probably because GDrive allows downloading files below a certain size

Reply:
Probably best to avoid this if we could. If the provided link intermittently fails, maybe try the huggingface location?

https://huggingface.co/datasets/samsum/blob/main/data/corpus.7z

```
filename = keras.utils.get_file(
    "corpus.7z",
    origin="https://huggingface.co/datasets/samsum/resolve/main/data/corpus.7z"
)
```